File: fs-examples/numbers.pl
#!/usr/bin/perl

open NUMBERS, '<', 'lotsanumbers.txt';

while (<NUMBERS>) {
    chomp;
    my $lineTotal = 0;
    my $lineCount = 0;
    foreach $number (split '\t') {
        $lineTotal += $number;
        $lineCount++;
    }
    print ($lineTotal/$lineCount);
    print ', ';
}
print "\n";

File: fs-examples/numbers.fs
"Read a file with multiple numbers per line, then print each line's sum"
dataFile := FSFile open:'lotsanumbers.txt'.

data := (dataFile readlines componentsSeparatedByString:'\t') doubleValue.

out println:((data @ \#+ ) / (data @ count)).

columns := (data @2 at: @ (((data objectAtIndex:0) count) iota)) .

out println:((columns @ \#+ ) / (columns @ count)).